File: Creational/Prototype/Java/products/ExtendedKeyboard.java
package patterns.Creational.Prototype.Java.products;

public class ExtendedKeyboard extends Keyboard
{
	/**
	 * @param int
	 * @param int
	 */
	public ExtendedKeyboard(final int keyQty, final int newNumPadQty)
	{
		super(keyQty);
		numPadQty = newNumPadQty;
	}

	/**
	 * @param ExtendedKeyboard
	 */
	public ExtendedKeyboard(final ExtendedKeyboard keyboard)
	{
		super(keyboard);
		this.numPadQty = keyboard.getNumPadQty();
	}

	/**
	 * @return int
	 */
	public int getNumPadQty()
	{
		return numPadQty;
	}

	/**
	 * @return Keyboard
	 */
	public ExtendedKeyboard clone()
	{
		return new ExtendedKeyboard(this);
	}

	private final int numPadQty;
}

File: Creational/Prototype/PHP/products/ExtendedKeyboard.php
<?php

namespace Creational\Prototype\PHP\products;

require_once('Keyboard.php');

class ExtendedKeyboard extends Keyboard
{
    /**
     * @param int $keyQty
     * @param int $newNumPadQty
     */
    public function __construct(int $keyQty, int $newNumPadQty)
    {
        parent::__construct($keyQty);
        $this->numPadQty = $newNumPadQty;
    }

    /**
     * @return int
     */
    public function getNumPadQty(): int
    {
        return $this->numPadQty;
    }

    /**
     * @return ExtendedKeyboard
     */
    public function clone(): Keyboard
    {
        return new self($this->getKeyQty(), $this->getNumPadQty());
    }

    private $numPadQty;
}
